libraries/extractor/src/main/java/androidx/media3/extractor/DtsUtil.java

```diff
@@ -149,6 +152,19 @@ public static int parseDtsAudioSampleCount(byte[] data) {
    * @return The number of audio samples represented by the syncframe.
    */
   public static int parseDtsAudioSampleCount(ByteBuffer buffer) {
+
+    // Check for sync or non sync word for DTS:X Profile 2.
+    // DTS:X Profile 2's FrameSize = 1024.
+    if ((buffer.getInt(0) == SYNC_FTOC_LE) ||
+        (buffer.getInt(0) == SYNC_FTOC_NON_SYNC_LE)) {
+      return 1024;
+    }
+    // Check for sync word for DTS Express.
+    // DTS Express's FrameSize = 4096.
+    else if (buffer.getInt(0) == SYNC_EXT_SUB_LE) {
+      return 4096;
+    }
```

Comment:
Could you please add the specification where suggests the frame sizes to the comment?

Reply:
@tianyif We are controlling the possible frame sizes from the encoder. Right now, these are the only possible frame sizes that are in use for legitimate streaming. The enforcement of these frame sizes are not listed in the specs.

Reply:
Yes, that would be appreciated! Thanks!

Reply:
@tianyif Done. Please see 5c84b879f3837b289fd7af9c70eb55078adb705b.